muspinsim/mpi.py

```diff
@@ -74,7 +75,7 @@ def broadcast_object(self, obj, selected_attrs=None):
             try:
                 v = self.comm.bcast(v, root=0)
             except OverflowError as exc:
-                raise OverflowError(f"Overflow error on: {k}, {v}") from exc
+                raise OverflowError(f"Overflow error on: {k}, {v}: {str(exc)}") from exc
```

Comment:
From having looked at the code, the logic we have for `write_error` means we can't rely on `raise ... from` since when we create our list of errors we stop raising and create a now error purely from text - hence the need for `str(exc)` when throwing errors that will be caught in `write_error`, like `LarkExpressionError`.

In this instance though, I don't see why we can't rely on `from`, as we don't seem to do any further handling with `OverflowError` that would break the chain of `raise ... from`. Having said that, there's no real harm in repeating the error message, so happy to take the cautious approach and use `from` and `str` here - just wanted to mention it in case I am missing some further logic we're doing?